pkg/cloud/services/elb/loadbalancer.go

```diff
@@ -144,7 +145,11 @@ func (s *Service) RegisterInstanceWithAPIServerELB(i *infrav1.Instance) error {
 	}
 
 	// Validate that the subnets associated with the load balancer has the instance AZ.
-	instanceAZ := s.scope.Subnets().FindByID(i.SubnetID).AvailabilityZone
+	isubnet := s.scope.Subnets().FindByID(i.SubnetID)
+	if isubnet == nil {
```

Comment:
Consider simplifying this code with

```
if subnet := s.scope.Subnets().FindByID(i.SubnetID); subnet == nil {
```

Reply:
Need to get AZ from subnet, so keep it out of if condition. Changed 'isubnet' to 'subnet'.